cpp/src/parquet/column_writer.cc

```diff
@@ -1206,7 +1206,7 @@ Status ConvertDictionaryToDense(const ::arrow::Array& array, MemoryPool* pool,
 }
 
 static inline bool IsDictionaryEncoding(Encoding::type encoding) {
-  return encoding == Encoding::PLAIN_DICTIONARY;
+  return encoding == Encoding::PLAIN_DICTIONARY || encoding == Encoding::RLE_DICTIONARY;
```

Comment:
Someone check me here. Is there some reason why this function wasn't considering `RLE_DICTIONARY` before? Or was it just an omission?

Reply:
There're some points:
1. https://github.com/apache/arrow/blob/main/cpp/src/parquet/encoding.cc#L444-L445 . `encoding` is not passed in Encoder
2. But shit, it's RLE in decoder 😅: https://github.com/apache/arrow/blob/main/cpp/src/parquet/encoding.cc#L1607
3. it will be detect and normalized in other place, like: https://github.com/apache/arrow/blob/main/cpp/src/parquet/column_reader.cc#L876

I'm not familiar with this when I previously working on this and I forgot this for a long time. I may unify them these days

Reply:
This is updated: https://github.com/apache/arrow/issues/41818